renderdoc/replay/entry_points.cpp

```diff
@@ -740,6 +742,127 @@ extern "C" RENDERDOC_API void RENDERDOC_CC RENDERDOC_EnumerateAndroidDevices(rdc
   *deviceList = ret;
 }
 
+bool endsWith(const string &value, const string &ending)
+{
+  if(ending.length() > value.length())
+    return false;
+
+  return (0 == value.compare(value.length() - ending.length(), ending.length(), ending));
+}
+
+bool installRenderDocServer(const string &deviceID)
+{
+  string targetApk = "RenderDocCmd.apk";
+  string serverApk;
+
+  // Check known paths for RenderDoc server
+  string exePath;
+  FileIO::GetExecutableFilename(exePath);
+  string exeDir = dirname(FileIO::GetFullPathname(exePath));
+
+  std::vector<std::string> paths;
+
+#if defined(RENDERDOC_APK_PATH)
+  string customPath(RENDERDOC_APK_PATH);
+  RDCLOG("Custom APK path: %s", customPath.c_str());
+
+  if(FileIO::IsRelativePath(customPath))
+    customPath = exeDir + "/" + customPath;
+
+  // Check to see if APK name was included in custom path
+  if(!endsWith(customPath, targetApk))
+  {
+    if(customPath.back() != '/')
+      customPath += "/";
+    customPath += targetApk;
+  }
+
+  paths.push_back(customPath);
+#endif
+
+  paths.push_back(exeDir + "/android/apk/" + targetApk);                         // Windows install
+  paths.push_back(exeDir + "/../share/renderdoc/android/apk/" + targetApk);      // Linux install
+  paths.push_back(exeDir + "/../../build-android/bin/" + targetApk);             // Local build
+  paths.push_back(exeDir + "/../../../../../build-android/bin/" + targetApk);    // macOS build
```

Comment:
Not being familiar at all with mac - I guess for the OS X install we can somehow include the apk in with the folder package thing, then add that path here?